### ISSUE TITLE
wrong detection with selector-combinator-space-before/after

### ISSUE THREAD

**Issue by tgfjt** [stylelint/stylelint#307]
Hi~ :sob: 

It seems to be wrong detection with checking `~` combinator.
Selectors are checked with a character on `styleSearch.js` so unfortunately `~=` is matched.

My code has `[attribute~=value]`, yes `~=`.
This is a sample code.

```css
.foo ~ .bar {
    color: black;
}

[m-label~=white] {
    color: white;
}
```

```bash
5:1 Expected single space after "~" combinator  (selector-combinator-space-after) [stylelint]
5:1 Expected single space before "~" combinator (selector-combinator-space-before) [stylelint]
```

**Comment by davidtheclark** [comment]
Thanks for uncovering that.

**Comment by jeddy3** [comment]
@davidtheclark Thanks for squashing this bug so quickly!

@tgfjt Released as `0.3.1` :)

**Comment by tgfjt** [comment]
@davidtheclark 
@jeddy3 
Wow very quick!:dash:  Thank you!
